microsoft/electionguard-cpp
# Swift (iOS app) demo using c interface

### Issue
https://github.com/microsoft/electionguard-core/issues/28

Fixes #28 

### Description
This PR adds a demo iOS app that shows how to use the electionguard-core library (via the C interface) to encrypt ballot selections.

The `ElectionGuard` struct contains a single static function `encryptSelection(objectId: String, vote: String)` that provides a Swift wrapper around the C functions for encrypting a plaintext selection and emulates the C demo app (see /apps/demo_in_c/main.c). 

### Testing
- Run `make build-ios` and copy the resulting `.a` binary into the root of the Xcode project. The library headers are already referenced directly.  
- Run the app on an iPhone or iPhone simulator and observe the output in the Xcode console. 

### Checklist
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.


💔Thank you!
